.werft/jobs/build/prepare.ts

```diff
@@ -104,26 +102,42 @@ function shouldCreateVM(config: JobConfig) {
 // createVM only triggers the VM creation.
 // Readiness is not guaranted.
 function createVM(werft: Werft, config: JobConfig) {
+    const cpu = config.withLargeVM ? 12 : 6;
+    const memory = config.withLargeVM ? 24 : 12;
+
+    // set some common vars for TF
+    // We pass the GCP credentials explicitly, otherwise for some reason TF doesn't pick them up
+    const commonVars = `GOOGLE_BACKEND_CREDENTIALS=${GCLOUD_SERVICE_ACCOUNT_PATH} \
+                        TF_VAR_dev_kube_path=${CORE_DEV_KUBECONFIG_PATH} \
+                        TF_VAR_harvester_kube_path=${HARVESTER_KUBECONFIG_PATH} \
+                        TF_VAR_preview_name=${config.previewEnvironment.destname} \
+                        TF_VAR_vm_cpu=${cpu} \
+                        TF_VAR_vm_memory=${memory}Gi \
+                        TF_VAR_vm_storage_class="longhorn-gitpod-k3s-202209251218-onereplica"`
+
     if (config.cleanSlateDeployment) {
         werft.log(prepareSlices.BOOT_VM, "Cleaning previously created VM");
-        VM.deleteVM({ name: config.previewEnvironment.destname });
+        // -replace=... forces recreation of the resource
+        exec(`${commonVars} \
+                        TF_CLI_ARGS_plan="-replace=harvester_virtualmachine.harvester" \
+                        ./dev/preview/workflow/preview/deploy-harvester.sh`,
+            {slice: prepareSlices.BOOT_VM}
+        );
     }
 
     werft.log(prepareSlices.BOOT_VM, "Creating  VM");
-    const cpu = config.withLargeVM ? 12 : 6;
-    const memory = config.withLargeVM ? 24 : 12;
-    VM.startVM({ name: config.previewEnvironment.destname, cpu, memory });
-    werft.currentPhaseSpan.setAttribute("preview.created_vm", true);
-}
 
-function applyLoadBalancer(option: { name: string }) {
-    function kubectlApplyManifest(manifest: string, options?: { validate?: boolean }) {
-        exec(`
-            cat <<EOF | kubectl --kubeconfig ${CORE_DEV_KUBECONFIG_PATH} apply --validate=${!!options?.validate} -f -
-${manifest}
-EOF
-        `);
+    const createVM = exec(`${commonVars} \
+                                    ./dev/preview/workflow/preview/deploy-harvester.sh`,
+        {slice: prepareSlices.BOOT_VM}
```

Comment:
I think this needs `dontCheckRc` if you want to check the status code in an if statement below. Alternatively you could also just try/catch it

```ts
try {
    const createVM = exec(`${commonVars} \
                                        ./dev/preview/workflow/preview/deploy-harvester.sh`,
            {slice: prepareSlices.BOOT_VM}
    );
} catch (err) {
        // NOTE: When you pass a slice to exec it already captures stdout and stdrr so you don't need to handle stderr here. 
        // createVM.stderr.split('\n').forEach(stderrLine => werft.log(prepareSlices.BOOT_VM, stderrLine))
        // NOTE: Using werft.fail so the error gets re-thrown - if we failed to create the VM I think we should fail the job, not just the slice
        werft.fail(prepareSlices.BOOT_VM, new Error(`Failed creating VM: {err}`))
}